spring-boot-admin-starter-client/src/main/java/de/codecentric/boot/admin/services/ApplicationFactory.java

```diff
@@ -0,0 +1,137 @@
+package de.codecentric.boot.admin.services;
+
+import de.codecentric.boot.admin.config.AdminClientProperties;
+import de.codecentric.boot.admin.model.Application;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.boot.actuate.autoconfigure.ManagementServerProperties;
+import org.springframework.boot.autoconfigure.web.ServerProperties;
+import org.springframework.boot.context.event.ApplicationReadyEvent;
+import org.springframework.context.event.EventListener;
+import org.springframework.web.context.WebApplicationContext;
+import org.springframework.web.util.UriComponentsBuilder;
+
+import java.net.InetAddress;
+import java.net.UnknownHostException;
+
+import static org.springframework.util.StringUtils.trimLeadingCharacter;
+
+/**
+ * A factory for creating instances of {@link Application}
+ *
+ * @author rfelgentraeger
+ */
+public class ApplicationFactory {
+
+
+    private AdminClientProperties client;
+
+    private ManagementServerProperties management;
+
+    private ServerProperties server;
+
+    private Integer serverPort;
+
+    private Integer managementPort;
+
+    @Value("${endpoints.health.path:/${endpoints.health.id:health}}")
+    private String healthEndpointPath;
+
+    public ApplicationFactory(AdminClientProperties client, ManagementServerProperties management, ServerProperties server) {
+        this.client = client;
+        this.management = management;
+        this.server = server;
+    }
+
+    public Application createApplication() {
+        //Remember, that not all information are set by the factory, e.g. id and status is set by ApplicationRegistry
+        //TODO: rethink about the co-existence of builder and factory pattern for the domain class Application (Appliation.Builder and ApplicationFactory)!
```

Comment:
I don't have a problem with having a builder and a factory.
The former is just for ease of development/reading code and the latter on having (multiple) strategies on how to construct an object.